### DISCUSSION TITLE
Data and content security question

### DISCUSSION THREAD

**Discussion by catedm** [wp-graphql/wp-graphql#2740] [Q&A]
Hey everyone,

I have a question about data security when using the plugin. Here is the problem I am facing:

- Our platform is a subscription based service that serves up paid content from WP graphql to a decoupled frontend application. WP is the CMS that houses our content via custom post types, and the frontend fetches the content from WP via WPGraphQL.
- Users are allowed to browse a limited version of the content before making a paid subscription.
- The problem is that - in the network response, our WPGraphQL calls return all of the content, including the premium paid content. So if someone wanted to steal our premium content, they could open up the developer tools, dig into the network calls to our WP site and get it from the `data` object that gets returned in the response before making a paid subscription.

Is there a way to hide the `data` object in the response using the plugin?

**Comment by justlevine** [comment]
Hey @catedm 

So in a pinch you can use the `graphql_return_modeled_data` [filter](https://www.wpgraphql.com/filters/graphql_return_modeled_data) or several other ones to restrict the return of data by your own conditiobak logic (including permission checks).

That said, WPGraphQL by default respects WordPress's built-in capabilities and permissions, so a better approach might be examining how those restricted CPTs are registered.
